avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroGenerator.java

```diff
@@ -49,7 +49,17 @@
          *
          * @since 2.7
          */
-        AVRO_BUFFERING(true)
+        AVRO_BUFFERING(true),
+
+        /**
+         * Feature that writes {@code char}, {@code char[]}, {@link Character}, and {@code Character[]} as {@code String strings} instead of
+         * the avro-standard integers.
```

Comment:
It appeared that when Jackson sees a `char[]`, it tries to write it out as an array of strings, and also tries to read it in as an array of strings. Avro instead uses an array of int for handling these, and while I adapted the schema generator, avro generator, and avro parser to handle this properly, I figured someone might want the old behavior still, so I tried to implement an option to restore the old behavior.